File: 1051-height-checker/main.go
func heightChecker(heights []int) int {
    
    n := len(heights)

    expected := make([]int, n)
    copy(expected, heights)

    sort.Slice(expected, func(i, j int) bool {
        return expected[i] < expected[j]
    })

    count := 0
    for i := range n {
        if heights[i] != expected[i] {
            count++
        }
    }

    return count
}

File: 1051-height-checker/main.js
/**
 * @param {number[]} heights
 * @return {number}
 */
var heightChecker = function(heights) {
    
    let n = heights.length;

    let expected = heights.slice();
    expected.sort(function(a, b) {return a - b});

    let count = 0;
    for (i = 0; i < n; i++) {
        if (heights[i] != expected[i]) {
            count++;
        }
    }

    return count;
};
